firmware/arduino/chessboard/uart.ino

```diff
@@ -0,0 +1,62 @@
+char incomingByte;
+char  buffer[5];
+int byteNum = -1; //-1 indicates that the start code hasn't been received
+
+//Wait for input
+void serialEvent2()
+{
+    //Loop through all available bytes
+    while (Serial2.available())
+    {
+        //Get byte
+        incomingByte = (char) Serial2.read();
+
+        //Reset buffer position
+        if (incomingByte == '~')
+        {
+            byteNum = 0;
+        }
+        //Add byte to buffer
+        else if (byteNum != -1)
+        {
+            buffer[byteNum] = incomingByte;
```

Comment:
This can be `buffer[byteNum++] = incomingByte` and then the last if statement can be removed